stratum/hal/lib/barefoot/bf_chassis_manager.cc

```diff
@@ -392,6 +428,40 @@ ::util::Status BFChassisManager::PushChassisConfig(
   return ::util::OkStatus();
 }
 
+::util::Status BFChassisManager::ApplyPortShapingConfig(
+    uint64 node_id, int unit, uint32 sdk_port_id,
+    const TofinoConfig::BfPortShapingConfig::BfPerPortShapingConfig&
+        shaping_config) {
+  switch (shaping_config.shaping_case()) {
+    case TofinoConfig::BfPortShapingConfig::BfPerPortShapingConfig::
+        kPacketShaping:
+      RETURN_IF_ERROR(bf_sde_interface_->SetPortShapingRate(
+          unit, sdk_port_id, true,
+          shaping_config.packet_shaping().max_burst_packets(),
+          shaping_config.packet_shaping().max_rate_pps()));
+      break;
+    case TofinoConfig::BfPortShapingConfig::BfPerPortShapingConfig::
+        kByteShaping:
+      // The SDE expects the rate in kbps.
+      RETURN_IF_ERROR(bf_sde_interface_->SetPortShapingRate(
+          unit, sdk_port_id, false,
+          shaping_config.byte_shaping().max_burst_bytes(),
+          shaping_config.byte_shaping().max_rate_bps() / 1000));
```

Comment:
Might be better to do this conversion in the wrapper

Reply:
I guess it comes down to this: Minimal wrapper around SDE or sane interface with QoL improvements. Keep in mind that bps will require `uint64`.